client/state/utils.js

```diff
@@ -23,6 +23,84 @@ export function isValidStateWithSchema( state, schema ) {
 }
 
 /**
+ * Creates a super-reducer as a map of reducers over keyed objects
+ *
+ * Use this when wanting to write reducers that operate
+ * on a single object as if it lived in isolation when
+ * really it lives in a map of similar objects referenced
+ * by some predesignated key or id. This could be used
+ * for example when reducing properties on a site object
+ * wherein we have many sites keyed by site id.
+ *
+ * Note! This will only apply the supplied reducer to
+ * the item referenced by the supplied key in the action.
+ *
+ * If no key exists whose name matches the given keyName
+ * then this super-reducer will abort and return the
+ * previous state.
+ *
+ * If some action should apply to every single item
+ * in the map of keyed objects, this utility cannot be
+ * used as it will only reduce the referenced item.
+ *
+ * @example
+ * const age = ( state = 0, action ) =>
+ *     GROW === action.type
+ *         ? state + 1
+ *         : state
+ *
+ * const title = ( state = 'grunt', action ) =>
+ *     PROMOTION === action.type
+ *         ? action.title
+ *         : state
+ *
+ * const userReducer = combineReducers( {
+ *     age,
+ *     title,
+ * } )
+ *
+ * export default keyedReducer( 'username', userReducer )
+ *
+ * dispatch( { type: GROW, username: 'hunter02' } )
+ *
+ * state.users === {
+ *     hunter02: {
+ *         age: 1,
+ *         title: 'grunt',
+ *     }
+ * }
```

Comment:
do you think it's worth removing this simple example whose purpose is to illustrate the behavior here? I felt like a tiny code snippet gave a lot more context than the summary description